### DISCUSSION TITLE
Tranglelab titan aero v6 clone doesn't fit without modifications.

### DISCUSSION THREAD

**Discussion by Gjuju** [1coderookie/KobraGoNeoInsights#22] [General]
Just to add some info's on the extruder replacement.

Mine was a junk, very noisy clicking sound on retraction, the motor shaft gear hole was to big, second black gear wrapped... 
A complete mess.

I ordered trianglelab titan aero v6 clone, it's another world.

But it doesn't fit out of the box
You must enlarge holes of the two hotend supports like on that picture. 2mm should be enough. 

![IMG_20231213_142841](https://github.com/1coderookie/KobraGoNeoInsights/assets/1523359/f9bbe794-9ea8-4196-887f-02f838ad37ee) 

The heatblock must be positioned parallel to X axis unless it's to close of the bottom wheel.

Great enhancements (for me though)!

Thanks a lot for all those informations, you made a really awesome job!

Gjuju

**Comment by 1coderookie** [comment]
Thanks :)

And thanks for showing your solution!  
Can you maybe add a picture of the final assembly as well?  
I'd then leave this discussion open and link to it from the site.. 

And iirc there was like a "left" and a "right" version of those TitanAeros, right? Which one do you have to get when you want to do this mod?

**Reply by Gjuju** [reply]
The one I bought is the trianglelab from aliexpress and it doesn't provide a right or left version.
The one they sell has motor on the left and extruder fan on the right like original.

Here is an assembled picture :
![IMG_20231216_133448](https://github.com/1coderookie/KobraGoNeoInsights/assets/1523359/3ea87983-bfc8-47df-9adc-d800cf3f1ab1)
![IMG_20231216_133454](https://github.com/1coderookie/KobraGoNeoInsights/assets/1523359/7e571fa8-44e0-4979-93e8-aaa695898647)
Feel free to use them on your website.

Best regards.

**Reply by 1coderookie** [reply]
Great, thanks!  

Btw: just saw your "GDSTIME" 5015 - does that one work with PWM?

**Reply by Gjuju** [reply]
Unfortunately not, BUT, strange behavior, it starts after a few seconds and works without noise at 30% THEN after 92%.

I'm finishing a simple electronic circuit for converting pwm to dc and driving any fan I want.

**Reply by 1coderookie** [reply]
Dang, I was hoping that one works with PWM - but thx for the info!  If you're using Klipper, you can try different cycle_times for that particular fan tho, as some work better with lower frequencies (like 100Hz) and others are working better with higher frequencies (like 20KHz).  
You can simply use a buck converter and adjust the voltage with a poti to adjust the spinning power of the fan (while PWM being set to 100% then), but that's nothing which would work automatically with the slicer settings. So it's a hack, but it's not as comfortable in the end like having a fan which works with PWM. 
Sunon should work, but I didn't try them yet, it's just what I came across.

**Reply by Gjuju** [reply]
I'm running marlin but I tried many freqs between 100Hz and 20kHz before (esp32 pwm + mosfet +24v psu), and get nothing really better with other frequencies.
20kHz is the original freq from main board.

As I'd like to keep a standard fan control over gcode, I'm working on / testing a very simple pwm to dc converter driving an generic xl4005 buck converter module.
Those are very cheap and have only 0.6v dropout.
It seems it woks well for now, if I'm satisfied I'll share it.
